samples/tutorials/mnist/02_Local_Development_with_Docker_Image_Components.ipynb

```diff
@@ -0,0 +1,548 @@
+    "])\n",
+    "\n",
+    "model.compile(optimizer='adam',\n",
+    "              loss='sparse_categorical_crossentropy',\n",
+    "              metrics=['accuracy'])\n",
+    "\n",
+    "print(model.summary())    \n",
+    "\n",
+    "mnist = tf.keras.datasets.mnist\n",
+    "(x_train, y_train),(x_test, y_test) = mnist.load_data()\n",
+    "x_train, x_test = x_train / 255.0, x_test / 255.0\n",
+    "\n",
+    "callbacks = [\n",
+    "  tf.keras.callbacks.TensorBoard(log_dir=bucket + '/logs/' + datetime.now().date().__str__()),\n",
+    "  # Interrupt training if val_loss stops improving for over 2 epochs\n",
+    "  tf.keras.callbacks.EarlyStopping(patience=2, monitor='val_loss'),\n",
+    "]\n",
+    "\n",
+    "model.fit(x_train, y_train, batch_size=32, epochs=5, callbacks=callbacks,\n",
+    "          validation_data=(x_test, y_test))\n",
+    "\n",
+    "\n",
+    "model.save(model_file)\n",
+    "\n",
+    "from tensorflow import gfile\n",
+    "\n",
+    "gcs_path = bucket + \"/\" + model_file\n",
+    "\n",
+    "if gfile.Exists(gcs_path):\n",
+    "    gfile.Remove(gcs_path)\n",
+    "\n",
+    "gfile.Copy(model_file, gcs_path)\n",
+    "with open('/output.txt', 'w') as f:\n",
+    "  f.write(gcs_path)\n",
+    "HERE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create docker file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now create a container that runs the script. Start by creating a `Dockerfile`. A `Dockerfile` contains the instructions to assemble a Docker image. The `FROM` statement specifies the Base Image from which you are building. `WORKDIR` sets the working directory. When you assemble the Docker image, `COPY` will copy the required files and directories (for example, `app.py`) to the filesystem of the container. `RUN` will execute a command (for example, install the dependencies) and commits the results. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create Dockerfile.\n",
+    "cat > ./tmp/components/mnist_training/Dockerfile <<EOF\n",
+    "FROM tensorflow/tensorflow:1.15.0-py3\n",
+    "WORKDIR /app\n",
+    "COPY . /app\n",
+    "EOF"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Build docker image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have created our Dockerfile we can create our Docker image. Then we need to push the image to a registry to host the image. \n",
+    "- We are going to use the `kfp.containers.build_image_from_working_dir` to build the image and push to the Google Container Registry (GCR), which makes use of [kaniko](https://cloud.google.com/blog/products/gcp/introducing-kaniko-build-container-images-in-kubernetes-and-google-container-builder-even-without-root-access).\n",
```

Comment:
to the Google Container Registry (GCR) -> to Container Registry (GCR)

Reply:
which makes use of -> which uses

Reply:
Modified as suggested.